Game manager rewrite (#122)
AmigaPorts/ACE: docs/programming/ace_in_a_nutshell.md

- lastly, _menuCommon_ should in its loop examine why its loop got called and
  decide what to do next. It can be done by e.g. examining some global variable.
  If it was because menuMapSelect requested launching game, you change state
  to _game_, otherwise you call `gameExit` which will effectively close the
  game.

> 

Author: It will be better to move it to separate issue.

- `genericLoop()` - called in a loop until game gets closed
- `genericDestroy()` - for freeing previously created managers

> If you preffer exiting your game in other ways than by calling `gameExit`, you

Reviewer: ```suggestion
If you prefer exiting your game in other ways than by calling `gameExit`, you
```

> can tune game main loop condition by defining `GENERIC_MAIN_LOOP_CONDITION`

Reviewer: And what is generic main.h? It's not mentioned until this point. This chapter is about basic game state concept, not about generic main.

Maybe move it to tutorial or describe it in generic.h as comment?

Author: This chapter as a whole literally explains everything about `main.h`... So... Where is the problem?

Reviewer: sorry, I was basing this on my memory and diffs, haven't looked at the whole text. You're right, it's good.

> before `#include`. For e.g.:

Reviewer: Use either 'For example', or 'E.g.' (exempli gratia)

``` c
#define GENERIC_MAIN_LOOP_CONDITION g_pGameStateManager->pCurrent
#include <ace/generic/main.h>
```

> At this point we changed generic main loop to check if there is any state in 

Reviewer: ```suggestion
This changes generic main loop's behavior to check if there is any state in 
```
